### PR TITLE
Fix #8229 [Bug] {{cloze:Text}} ignores repeated words and just check first of them

### LINKED ISSUE #8229: [Bug] {{cloze:Text}} ignores repeated words and just check first of them
AnkiDroid ver 2.14.6
1. in card filed I use "This is {{c1::test}} which is containing {{c1::test}} word twice"
2. in Front and Back template I have "{{cloze:Field_name}}"
3. now when I Type answer like "test, test" in front and tap the answer:

I expect this: 
test, test (Check mark)

but I wee this:
test, test (first highlighted in green and second in red followed by downside arrow on next line and after that shows answer like the following line)
test (one time, so It recognize it as false answer and there is no check button)

### PR DESCRIPTION
**Pull request by LittleFish33**
## Purpose / Description
Fix #8229 [Bug] {{cloze:Text}} ignores repeated words and just check first of them

## Fixes
Fixes  #[8229](https://github.com/ankidroid/Anki-Android/issues/8229)

## Approach
In the original file, the project uses `set<String>` to save the correct answer to use for {{type::cloze::NN}} fields (refer to the `private String contentForCloze(String txt, int idx) ` function in AbstractFlashcardViewer.java). In this case, {{cloze:Text}} ignores repeated words because there cannot be duplicate elements in a set. Therefore, I change the type from `set<String>` to `ArrayList<String>`

## How Has This Been Tested?

working well on my pixel phone

## Learning (optional, can help others)
_Describe the research stage_

_Links to blog posts, patterns, libraries or addons used to solve this problem_

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java [hunk 1]
```diff
@@ -827,9 +827,9 @@ private String contentForCloze(String txt, int idx) {
         @SuppressWarnings("RegExpRedundantEscape") // In Android, } should be escaped
         Pattern re = Pattern.compile("\\{\\{c" + idx + "::(.+?)\\}\\}");
         Matcher m = re.matcher(txt);
-        Set<String> matches = new LinkedHashSet<>(); // Size can't be known in advance
-        // LinkedHashSet: make entries appear only once, like Anki desktop (see also issue #2208), and keep the order
-        // they appear in.
+        ArrayList<String> matches = new ArrayList<>(); // Size can't be known in advance
+        // changes the matches type from Set<String> to ArrayList<String>, fix issue #8229
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java:831]: My main concern here is that there's a comment stating that the opposite was intended. Using a set was a conscious decision, and we don't understand why this happened, or why requirements have changed.

Could we:

* Get a link to the Anki source where this is implemented/changed so a set is now the correct decision
* Add a Unit Test in a separate commit - so people won't immediately revert this if there is an overlooked problem, and instead will find a solution which fixes both issues

Comment (LittleFish33) [reply AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java:831]: Thank you for your comments. I will try to work on this problem.

Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java:831]: Let us know when it's done

Comment (LittleFish33) [reply AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java:831]: I have tried to use the template in issue #8229, and I get the same results in both Desktop: 2.1.35 and Android: 2.14.6 (the same version in the issue). Both the Desktop and Android require to type test only once. This is contrary to the statement in issue #8229. Therefore, I've asked the questioner to check the card fields. I think the main reason for using a `set` here is to be consistent with the desktop version.

**Pull request closed without merging**